src/ttsim/tt/shared.py

```diff
@@ -66,28 +66,30 @@ def join(
     -------
     The joined array.
     """
-    # For each foreign key and for each primary key, check if they match
-    matches_foreign_key = foreign_key[:, None] == primary_key
-
-    # For each foreign key, add a column with True at the end, to later fall back to
-    # the value for unresolved foreign keys
-    padded_matches_foreign_key = xnp.pad(
-        matches_foreign_key,
-        ((0, 0), (0, 1)),
-        "constant",
-        constant_values=True,
+    # First, get the sort order of primary_key to enable efficient lookup
+    sort_indices = xnp.argsort(primary_key)
+    sorted_primary_key = primary_key[sort_indices]
+    sorted_target = target[sort_indices]
+
+    # Use searchsorted to find where each foreign_key would be inserted
+    # in the sorted primary_key array
+    positions = xnp.searchsorted(sorted_primary_key, foreign_key, side="left")
+
```

Comment:
```suggestion
    # Find where each foreign_key would be inserted in the sorted primary_key array
    positions = xnp.searchsorted(sorted_primary_key, foreign_key, side="left")

```

Don't state the obvious.